### PR TITLE
Updating and fixing unit tests for DirectoryServices.AccountManagement

### PR DESCRIPTION
**Pull request by FWest98**
# Description

While looking into the code for S.DS.AM, I found that the unit tests were not written consistently. The ones in `AccountManagementTests` were written in a modern fashion, using the `LdapConfiguration` class to provide dynamic access to various LDAP/AD services, which works very nice. These tests seem to be well-designed.

However, the other tests were older and were partially broken. They had credentials hardcoded, and, for example, the `GroupPrincipalTest::IsMemberOfTest` method actually never called the `CreateManyUsersInGroup` method that would set up the data for that test. Other tests would simply throw exceptions. In practice, if no connection to the domain controller was set up, the tests would not run without any notice.

I found reference to a previous discussion about these tests: https://github.com/dotnet/corefx/pull/21522/files#r124152887. They were called "pretty bogus" and there were plans to replace them, however, I think that never happened.

In fact, most functionality that was supposed to be tested in these old tests is actually already covered by tests in `AccountManagementTests`. So to clean up a bit, for the remaining functionality, I added new tests that follow the new style, and I removed the old tests.

# Customer Impact
I think N/A

# Testing
All new tests work locally. I believe that at the moment, none of these tests are actually being run in the "normal" CI pipelines, since it doesn't look like it has an AD server set up. I don't know if they are being run in any other setting. This might be something to investigate.

# Risk
I don't think there is a lot of risk; this PR is just about tests so it does not impact the shipped product. The only risk I can imagine is that the old tests were actually run in some setting other than in the CI pipelines I could find, and that the new tests do not fully cover all scenarios, after which there is a risk that something will break and will go unnoticed. However, I believe with the additions in this PR, the new tests have at least the same coverage as the old tests, but with better quality and maintainability.

### CODE REVIEW

File: src/libraries/System.DirectoryServices.AccountManagement/tests/ComputerPrincipalTest.cs
Comment (joperezr) [inline src/libraries/System.DirectoryServices.AccountManagement/tests/ComputerPrincipalTest.cs:48]: What about this test? this one doesn't seem to have been migrated over to AccountManagementTests right?

File: src/libraries/System.DirectoryServices.AccountManagement/tests/PrincipalTest.cs
Comment (joperezr) [inline src/libraries/System.DirectoryServices.AccountManagement/tests/PrincipalTest.cs:37]: Looks like we are missing all of the base tests as well. I don't see these getting migrated over, only the ctor tests seemed to be moved.

File: src/libraries/System.DirectoryServices.AccountManagement/tests/ComputerPrincipalTest.cs
Comment (FWest98) [reply src/libraries/System.DirectoryServices.AccountManagement/tests/ComputerPrincipalTest.cs:48]: That's included here if I'm correct:
https://github.com/dotnet/runtime/blob/0ea9425bcf527b615a07f6cf5fe9936b6070cc0a/src/libraries/System.DirectoryServices.AccountManagement/tests/AccountManagementTests.cs#L22-L28
You could argue that not the exact same constructor signature was used, but that does not matter for this specific check, which is about the context.

File: src/libraries/System.DirectoryServices.AccountManagement/tests/PrincipalTest.cs
Comment (FWest98) [reply src/libraries/System.DirectoryServices.AccountManagement/tests/PrincipalTest.cs:37]: Most of them were already part of the new test suite:
- `AddExistingPrincipal`
-- Tests adding a principal - already in new suite: https://github.com/dotnet/runtime/blob/0ea9425bcf527b615a07f6cf5fe9936b6070cc0a/src/libraries/System.DirectoryServices.AccountManagement/tests/AccountManagementTests.cs#L102
-- Tests adding duplicate principal - already in new suite: https://github.com/dotnet/runtime/blob/0ea9425bcf527b615a07f6cf5fe9936b6070cc0a/src/libraries/System.DirectoryServices.AccountManagement/tests/AccountManagementTests.cs#L292
- `TestExtendedPrincipal` I pulled apart from this PR into another PR for a bugfix (#56696), as writing those tests properly causes a bug to trigger that I fixed there. I did not want to bring in breaking tests here and have that PR depend on this one. I could move things around a bit though, if desired.
- `Save_ThrowsInvalidOperationException` tries to store a principal using an "incorrect" context (different from the creating context as well); which I migrated to https://github.com/dotnet/runtime/blob/0ea9425bcf527b615a07f6cf5fe9936b6070cc0a/src/libraries/System.DirectoryServices.AccountManagement/tests/AccountManagementTests.cs#L319

**Comment by joperezr** [comment]
cc: @tquerec @grubioe can you please take a look at this one too?

**Comment by danmoseley** [comment]
BTW, thank you for working on these tests @FWest98 it is good to have some attention on these old tests.

**Comment by jeffhandley** [comment]
> cc: @tquerec @grubioe can you please take a look at this one too?

I've marked this as blocked until we get a review from @tquerec and/or @grubioe.

**Comment by grubioe** [comment]
@BRDPM and @jay98014  - can you review?

**Comment by BRDPM** [comment]
I've looked over these but I would like some backup before we approve - I'll set up some time with @jay98014 to go over these next week.

**Review by joperezr — APPROVED** [review]
Changes here look good, and seems like Windows-DirectoryServices team signed off too. Will ammend the last commit to force a rerun of CI and merge if CI is green.

**Comment by danmoseley** [comment]
Thanks @FWest98

**Pull request merged**
